### ISSUE TITLE
Support Jakarta EE 10 including major dependency updates (spring boot 3.3, camel 4.4, CXF 4.1)

### ISSUE THREAD

**Issue by Thopap** [oehf/ipf#414]
As part of this issue, IPF will be updated to the Jakarta EE 10, which also require certain other major dependency updates.

* Migrate from javax -> jakarta (JEE 10)
* Java 17 as minimum
* Spring Framework 6
* Spring Boot 3
* CXF 4.1 (waiting for https://issues.apache.org/jira/browse/CXF-8671)
* Camel 4.4 (planned for may 2023: https://camel.apache.org/blog/2023/01/camel4roadmap/ )
* Tomcat 10
* HerasAF 3.0.2 (https://github.com/prolutionsGmbH/herasaf-xacml-core/issues/13)
* Hapi FHIR 7.x (https://github.com/hapifhir/hapi-fhir/issues/2082)

**Comment by ohr** [comment]
There is a blocker so far as hapi-fhir is still on javax-based JEE version. There is increasingly interest documented in https://github.com/hapifhir/hapi-fhir/issues/2082, but no hapi-fhir release implementing this is currently planned for

**Comment by ohr** [comment]
Another issue is HL7v3 request/response translation from IHE Gazelle JAXB models (net.ihe.gazelle:hl7v3-code-jar and its transitive dependencies) that are also annotated with the old XmlBind annotations

**Comment by Thopap** [comment]
For heras-xacml-core a new version 3.0.2 with  J2EE 10 Jaxb compatiblity is now available.

**Comment by kiranam786** [comment]
Hi @Thopap, We are using IPF's latest version i.e., 4.7.0. We wanted to upgrade our projects to spring 6 but due to incompatibility with IPF modules we could not do it. I could see there are couple of mile stones.
Is there any timeline that we have for the release of  IPF 5.0 ?

**Comment by ohr** [comment]
We mainly depend on https://github.com/hapifhir/hapi-fhir/issues/2082 to be fixed, which unfortunately has an unknown timeline. Camel 4.0 should have a release, too, but this can be expected soon. 

Some HL7v3 model library is also stuck on javax, but I don't expect this to change anytime soon; so I would rather remove this library.

**Comment by unixoid** [comment]
@cyrillzadra No worries, this library is not essential for HL7v3 components.  Moreover, we need these components by ourselves and would not remove them :-)

**Comment by qligier** [comment]
HAPI FHIR has completed the migration to Jakarta:

> HAPI FHIR 6.11.5-SNAPSHOT and later are using the new Jakarta APIs. This will be released as HAPI FHIR 7.0.0 in Feb 2024.

**Comment by ohr** [comment]
HAPI FHIR has merged back the migration into its develop build, but there are no tags yet we could consume.
I keep the [feature/414-jee10](https://github.com/oehf/ipf/tree/feature/414-jee10) branch updated with all the other stuff, but I have left all FHIR libraries disabled there.

Regarding Gazelle HL7v3 model library (hl7v3-code): this library and its dependencies don't come with any licensing information, so have to remove it from the IPF distribution. A license is a grant of rights. No license - no rights to use.

**Comment by Thopap** [comment]
For documentation purpose:

CXF 4.0 + IPF 5 is basically working fine with the exception:
In Case a async webservice responder endpoint is used + Camel Message Logging is enabled, CXF will run into a NullPointer in Netty because CXF preserve the HTTP Servlet Request object even if the HTTP request / response is already done.

In IPF one testcase was affected (testIti56 ) which was currently fixed by disable the Message Logging (https://github.com/oehf/ipf/commit/84c8591f4d8acceb396f94b3ad06f9606074d06e)

So basically CXF 4.1 will fix this handling:
https://issues.apache.org/jira/browse/CXF-9004

(Further details also here https://lists.apache.org/list?users@cxf.apache.org:2024-7)

**Comment by rbastiaansen732** [comment]
Great work, all! 

I'm testing our software in combination with this branch. The major upgrade is looking good.

**Comment by rbastiaansen732** [comment]
CXF 4.1 https://github.com/apache/cxf/releases/tag/cxf-4.1.0 resolved https://issues.apache.org/jira/browse/CXF-8671 cool! 🥳
